sdks/python/apache_beam/io/gcp/datastore/v1new/types_test.py

```diff
@@ -47,30 +49,52 @@ def setUp(self):
         # Don't do any network requests.
         _http=mock.MagicMock())
 
+  def _assert_keys_equal(self, beam_type, client_type, expected_project):
+    self.assertEqual(beam_type.path_elements[0], client_type.kind)
+    self.assertEqual(beam_type.path_elements[1], client_type.id)
+    self.assertEqual(expected_project, client_type.project)
+
   def testEntityToClientEntity(self):
+    # Test conversion from Beam type to client type.
     k = Key(['kind', 1234], project=self._PROJECT)
     kc = k.to_client_key()
-    exclude_from_indexes = ('efi1', 'efi2')
+    exclude_from_indexes = ('datetime', 'key')
     e = Entity(k, exclude_from_indexes=exclude_from_indexes)
-    ref = Key(['kind2', 1235])
-    e.set_properties({'efi1': 'value', 'property': 'value', 'ref': ref})
+    properties = {
+      'datetime': datetime.datetime.utcnow(),
+      'key_ref': Key(['kind2', 1235]),
+      'bool': True,
+      'float': 1.21,
+      'int': 1337,
+      'unicode': 'text',
+      'bytes': b'bytes',
+      'geopoint': GeoPoint(0.123, 0.456),
+      'none': None,
+      'list': [1, 2, 3],
+      'entity': Entity(Key(['kind', 111])),
+      'dict': {'property': 5},
+    }
+    e.set_properties(properties)
     ec = e.to_client_entity()
     self.assertEqual(kc, ec.key)
     self.assertSetEqual(set(exclude_from_indexes), ec.exclude_from_indexes)
     self.assertEqual('kind', ec.kind)
     self.assertEqual(1234, ec.id)
-    self.assertEqual('kind2', ec['ref'].kind)
-    self.assertEqual(1235, ec['ref'].id)
-    self.assertEqual(self._PROJECT, ec['ref'].project)
-
-  def testEntityFromClientEntity(self):
```

Comment:
We don't need this test anymore ?

Reply:
I merged the two tests together (see "Test reverse conversion" below).